src/torrent.ts

```diff
@@ -48,8 +50,48 @@ export enum SnatchError {
 export async function parseTorrentFromFilename(
 	filename: string,
 ): Promise<Metafile> {
-	const data = await readFile(filename);
-	return Metafile.decode(data);
+	return Metafile.decode(await readFile(filename));
+}
+
+export async function parseTorrentWithMetadata(
+	filename: string,
+	torrentInfos: TorrentMetadataInClient[],
+): Promise<Metafile> {
+	const meta = await parseTorrentFromFilename(filename);
+	const client = getClient();
+	if (client?.type === Label.QBITTORRENT) {
```

Comment:
why is qBittorrent special cased? the idea of being able to rely more on public APIs and less on private APIs sounds great in theory. Is there a hard technical limitation?

Reply:
qBit strips the trackers from the .torrent files in torrentDir. Getting it from client requires an individual request for each torrent, the normal get all torrents request only returns the active one (none if paused). Luckily, the trackers are in the .fastresume files that are also in torrentDir. It also contains the category and tag so we just read for there.

qBit does have a fallback if users are passing the files through the cross-seed arg.

Reply:
I appreciate that we have a fallback that doesn't rely on private APIs. We can accept this tradeoff for now, but I think we should think about our path to supporting SQLite resume data. I'm really hoping that a lot of the performance issues with making API calls to qBittorrent go away but I haven't used it. 

In theory, making N+1 calls should be viable enough if we combine it with lazy loading of the relevant data and small searchLimits (100-200 api calls to localhost _should_ be basically nothing).